VultisigApp/VultisigApp/View Models/VaultDefaultCoinService.swift

```diff
@@ -39,6 +39,10 @@ class VaultDefaultCoinService {
             for coin in coins {
                 self.context.insert(coin)
                 vault.coins.append(coin)
+
+                Task {
+                    try await CoinSelectionViewModel().addDiscoveredTokens(nativeToken: coin, to: vault)
```

Comment:
This is weird , create a `CoinSelectionViewModel()` here ,

Reply:
I was also confused, but then I could see this service is in the same folder as all view models, so maybe we can create a service responsible for adding all coins and removing this view model.